### PR TITLE
Megafauna and necropolis tendrils are more common

### PR DESCRIPTION
**Pull request by ChangelingRain**
Megafauna have, initially, a 50% chance of bubblegum, 33%~ chance of ash drakes, and 16%~ chance of colussus. After bubblegum is spawned, that changes to 66%~ chance of ash drakes and 33%~ chance of colossus.

Tweaked limiters on how close things can spawn to each other:
Nothing can spawn within 7 tiles of megafauna
Tendrils can spawn slightly closer to each other(but not close enough a collapse can hit another tendril)
Standard hostile mobs can show up next to tendrils/tendrils can show up next to standard hostile mobs.

These will ggggenerally increase how many things spawn due to less stringent limitations(but I removed the extra spawn changes)

### CODE REVIEW

**Comment by GunHog** [comment]
RIP MINERS!!

**Comment by FantasticFwoosh** [comment]
You are aware that goldgrub replication means that you can farm them in xenobio with a lazarus, then with plasma forcefeed them from xenobio dark purple slime extracts?

Now we just need to re-establish goldgrub loot (it doesn't drop it because it can't be butchered, probably a bug because it can only be ashwalker gibbed) and you have grub ranching.

**Comment by RemieRichards** [comment]
>Nothing can spawn within 7 tiles of megafauna(but megafauna can and will spawn within 7 tiles of something)

so... the check is pointless?
the only goal of preventing things spawning within 7 tiles of a megafauna is to stop megafauna and normal mobs being grouped together, yet allowing the other case... allows the grouping?

**Comment by ChangelingRain** [comment]
It's not pointless, because it does decrease mob density.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 1]
```diff
@@ -165,19 +165,19 @@
 
 /mob/living/simple_animal/hostile/asteroid/goldgrub/proc/EatOre(atom/targeted_ore)
 	for(var/obj/item/weapon/ore/O in targeted_ore.loc)
-		if(loot.len < 10)
-			loot += O.type
+		if(is_type_in_list(O, wanted_objects))
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: ```wanted_objects``` should be a typecache.
why isn't it a typecache yet.

**Comment by GunHog** [comment]
@WJohn You should submit your request for revaming the monster spawn system. RR and Joan are more than capable of pulling that off.

**Comment by RemieRichards** [comment]
@ChangelingRain if only it were 2 files simple, there's cases where ```wanted_objects``` is "reset" into a raw ```/list()``` with no assoc keys, and I know for a fact (cos I wrote it) that mecha pilots modify their ```wanted_objects``` list a lot, there aren't too many uses of ```wanted_objects``` in the code, but it's more than 2 files.

**Comment by KorPhaeron** [comment]
Not sure about this

**Comment by Ergovisavi** [comment]
:+1: On more megafauna
:-1: On more tendrils

Tendrils are already incredibly plentiful, and their destruction results in permanent instakill chasms all over the place. Pls no.

**Comment by Durkel** [comment]
I thought the point of mining was to get minerals for the station, not play 2d monster hunter.

**Comment by Ergovisavi** [comment]
The point of mining is to create an entertaining experience

How it goes about that is irrelevant

**Comment by TechnoAlchemisto** [comment]
The last thing mining needs is more megafauna spawns, especially with their tendency of spawning close to the base. 

More tendrils is okay, I guess, but not really needed.

**Comment by ma44** [comment]
TFW you think a infinite supply points loophole is going to get merged expect it's removed

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -261,9 +261,14 @@
 
 ////////////////////////////////End Gibtonite
 
+#define SPAWN_MEGAFAUNA "bluh bluh huge boss"
```
Comment (francinum) [inline code/modules/mining/mine_turfs.dm]: No, bad.

**Comment by Kromgar** [comment]
I like the more spawning of tendrils but twice as many megabeasts how about 20-30% higher chance of megabeasts

**Comment by ChangelingRain** [comment]
Adjusted;
The percentage of megafauna now goes up from 4.16%~ of spawns to 5.84%~ of spawns
The percentage of tendrils now goes up from 5.55%~ of spawns to 9.74%~ of spawns

I'm not sure this is actually accurate because pickweight() is arcane and might be biased.

**Comment by francinum** [comment]
GOOD JOB.

**Comment by ma44** [comment]
9.74 / 5.55 = 1.75~
tendril spawn rate increased by an average of 75%
tendril max mobs
0.75 x 3 = 2.25 
literally up to 225% more mobs can spawn

**Comment by ChangelingRain** [comment]
Testing in its current state indicates it works... relatively well, and doesn't spawn more than 10 megafauna on average, though it will spawn more than 10 tendrils most of the time(below 20, usually)

**Comment by TechnoAlchemisto** [comment]
10 megafauna is way too many, why on earth would you want that many

**Comment by KorPhaeron** [comment]
There are already too many for basically every player outside of ergo

**Comment by ChangelingRain** [comment]
lzimann requested more megafauna

**Comment by GunHog** [comment]
**THIS IS NOT OKAY**

screenshot 2016-10-05 16 47 10

**Comment by ChangelingRain** [comment]
that's not a recent shot aaaaaa

**Comment by KorPhaeron** [comment]
jesus christ

File: code/modules/mining/mine_turfs.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/mining/mine_turfs.dm]: Fix the define

**Comment by optimumtact** [comment]
Stale, you need to work with Kor to get this to a state where they're happy.

**Comment by ChangelingRain** [comment]
I removed the increased spawn chances(but this will probably still increase spawns a little on its own and will definitely increase megafauna, because of no failed bubblegum spawns)

**Comment by GunHog** [comment]
It is cool as long as you do not have literally 5 tendrils in the same screenshot!

**Comment by optimumtact** [comment]
@KorPhaeron are you happy?

**Comment by KorPhaeron** [comment]
yeah sure fine

File: code/game/turfs/simulated/floor/plating/asteroid.dm [hunk 3]
```diff
@@ -144,11 +144,14 @@
 	turf_type = /turf/open/floor/plating/asteroid/airless
 
 
-
+#define SPAWN_MEGAFAUNA "bluh bluh huge boss"
```
Comment (optimumtact) [inline code/game/turfs/simulated/floor/plating/asteroid.dm]: SHIT

Comment (ChangelingRain) [inline code/game/turfs/simulated/floor/plating/asteroid.dm]: IF THIS FAILS I WANT IT TO RUNTIME NOBODY IS EVER GONNA SEE IT DON'T WORRY

**Pull request merged**